lib/cstruct.mli

```diff
@@ -181,6 +181,16 @@ val of_string: ?allocator:(int -> t) -> string -> t
     with the underlying buffer allocated by [alloc]. If [allocator] is not
     provided, [create] is used. *)
 
+(** {2 Comparison } *)
+
+val equal : t -> t -> bool
+(** [equal t1 t2] is [true] iff [t1] and [t2] correspond to the same sequence of
+    bytes. *)
+
+val compare : t -> t -> int
+(** [compare t1 t2] is [-1] if [t1] precedes [t2] by lexicographical order, [1]
+    if [t2] precedes [t1], and [0] otherwise. *)
```

Comment:
Le lundi, 2 février 2015 à 22:28, David Kaloper a écrit :
> Yeah, if it's only a consistent total order for Map.Make and the like, it could be made slightly faster...
> I'm undecided. Third-party ideas?

Note that Pervasives polymorphic comparison is defined just like that (while it happens to sort tuples lexicographically that's actually not guaranteed by the interface spec.)

Best,

Daniel